rehack_tests/stdlib/stdlib.cma.php/Genlex.php

```diff
@@ -319,83 +319,120 @@ private static function load($joo_global_object) {
+                    case 62:
+                    // FALLTHROUGH
+                    case 63:
+                    // FALLTHROUGH
+                    case 64:
+                      $switch__0 = 2;
+                      break;
+                    // FALLTHROUGH
+                    default:
+                      $switch__0 = 0;
                     }
-                    throw $runtime["caml_wrap_thrown_exception"](Vector{0, $Stream[2], $cst__0}) as \Throwable;
-                  }
-                  else if ($switcher === 40) {
-                    $call1($Stream[12], $strm);
-                    if ($counter < 50) {
-                      $counter__0 = (int) ($counter + 1);
-                      return $maybe_comment->contents($counter__0, $strm);
-                    }
-                    return $caml_trampoline_return(
-                      $maybe_comment->contents,
-                      varray[0,$strm]
-                    );
-                  }
-                  else if ($switcher === 45) {
-                    $call1($Stream[12], $strm);return $neg_number($strm);
-                  }
-                  else if ($switcher === 9) {}
-                  else if ($switcher === 10) {}
-                  else if ($switcher === 12) {}
-                  else if ($switcher === 13) {}
-                  else if ($switcher === 26) {}
-                  else if ($switcher === 32) {
-                    $call1($Stream[12], $strm);continue;
-                  }
-                  else if ($switcher === 48) {}
-                  else if ($switcher === 49) {}
-                  else if ($switcher === 50) {}
-                  else if ($switcher === 51) {}
-                  else if ($switcher === 52) {}
-                  else if ($switcher === 53) {}
-                  else if ($switcher === 54) {}
-                  else if ($switcher === 55) {}
-                  else if ($switcher === 56) {}
-                  else if ($switcher === 57) {
-                    $call1($Stream[12], $strm);
-                    $reset_buffer(0);
-                    $store($l);
-                    return $number($strm);
-                  }
-                  else if ($switcher === 33) {}
-                  else if ($switcher === 35) {}
-                  else if ($switcher === 36) {}
-                  else if ($switcher === 37) {}
-                  else if ($switcher === 38) {}
-                  else if ($switcher === 42) {}
-                  else if ($switcher === 43) {}
-                  else if ($switcher === 47) {}
-                  else if ($switcher === 58) {}
-                  else if ($switcher === 60) {}
-                  else if ($switcher === 61) {}
-                  else if ($switcher === 62) {}
-                  else if ($switcher === 63) {}
-                  else if ($switcher === 64) {$switch__0 = 2;break;
-                  }
-                  else {$switch__0 = 0;}
+                  if ($continue_label === "#") {continue;}
```

Comment:
Why not just have the exact name of the label that the switch wants to continue to? `continue_label="a"` - it wasn't clear that it was even possible for a switch to continue to anything but the unlabeled `"#"`, but I think it needs to be supported, so why make a special `#` to represent "jump out of the switch and then continue to a label-less for loop when we exit the switch". It makes me wonder how the switch would instruct us to jump out of the switch and into a labeled loop like `a:`.

Reply:
Using the label of the loop that contains the switch when replacing "continue;" statements inside the loop would be possible, but would require retaining the label in input. (Maybe with a Switch(string) case in input.enclosed_by). I can do that--it might make the input cleaner in certain cases.

But if the switch isn't enclosed with a labeled loop, then we'd still need some kind of indication to continue after breaking out of the loop. I used "#" to denote this case. I suppose an alternative could be to use "" in that case instead, since I effectively treated "" as the label in unlabeled loops elsewhere. Then there wouldn't be a need for "#" at all.

Reply:
I started to try removing "#", but it's a little more complicated than I thought. It might allow us to clean up more things, but I'd prefer to land this now and do that clean up later.